addons/web/static/src/views/fields/properties/properties_field.js

```diff
@@ -410,6 +390,63 @@ export class PropertiesField extends Component {
             propertyDefinition.name = newName;
         }
     }
+
+    /**
+     * Open the popover with the property definition.
+     *
+     * @param {DomElement} target
+     * @param {string} propertyName
+     * @param {boolean} isNewlyCreated
+     */
+    _openPropertyDefinition(target, propertyName, isNewlyCreated = false) {
+        const propertiesList = this.propertiesList;
+        const propertyIndex = propertiesList.findIndex(
+            (property) => property.name === propertyName
+        );
+
+        // maybe the property has been renamed because the type / model
+        // changed, retrieve the new one
+        const currentName = (propertyName) => {
```

Comment:
Can't you somehow keep a mapping of "old" propertyName -> "updated" propertyName ? Not knowing all details so maybe not doable, just proposing to avoid an O(n²) construct and going back into an O(n) if we only have to find a property based on a "always correct" naming.